components/collator/src/options.rs

```diff
@@ -198,6 +202,27 @@ pub enum CaseFirst {
     UpperFirst = 2,
 }
 
+impl From<CollationCaseFirst> for CaseFirst {
+    fn from(value: CollationCaseFirst) -> Self {
+        match value {
+            CollationCaseFirst::Upper => CaseFirst::UpperFirst,
+            CollationCaseFirst::Lower => CaseFirst::LowerFirst,
+            CollationCaseFirst::False => CaseFirst::Off,
+            _ => unreachable!("`CaseFirst` and `CollationCaseFirst` must always be in sync."),
+        }
+    }
+}
+
+impl From<CaseFirst> for CollationCaseFirst {
+    fn from(value: CaseFirst) -> Self {
+        match value {
+            CaseFirst::Off => CollationCaseFirst::False,
+            CaseFirst::LowerFirst => CollationCaseFirst::Lower,
+            CaseFirst::UpperFirst => CollationCaseFirst::Upper,
+        }
```

Comment:
Why you don't need a catch case despite CaseFirst being non-exhaustive?

Should we switch Options to use CollactionCaseFirst from Preferences then?

Reply:
`#[non_exhaustive]` only applies for external crates that use the enum.

I preserved the old enum because I wanted to avoid unnecessary breaking changes, but we can switch to that if preferred.

Reply:
We have a shot at doing it in 2.0, I think i'd prefer that. @sffc @Manishearth @robertbastian  - you ok with that?

Reply:
I have a slight preference for removing the settings from `options` and moving them to `preferences` in 2.0

Reply:
Same.

Reply:
Just to be clear – we would move the relevant options to `CollatorPreferences`, using the structs on `icu_locale_core`, and the rest of the options would still be in `CollatorOptions`, is that right?

Reply:
correct!